src/modals/CreateSponsorshipModal.tsx

```diff
@@ -154,7 +169,10 @@ function CreateSponsorshipModal({ balance: balanceProp, onResolve, ...props }: P
                         onBlockNumber: waitForIndexedBlock,
                     })
 
-                    onResolve?.(sponsorshipId)
+                    onResolve?.({
+                        sponsorshipId,
+                        streamId: streamIdProp,
```

Comment:
This should be the most recent `streamId` (`formData.streamId`). Otherwise if the modal hasn't been given an id the resolved `streamId` will be `undefined`, no?